src/common/CMakeLists.txt

```diff
@@ -18,7 +18,9 @@ endif()
 
 
 shared_add_library(sfscommon ${COMMON_SOURCES})
-shared_target_link_libraries(sfscommon ${YAML_CPP_LIBRARIES} STATIC sfserr SHARED sfserr_pic)
+shared_target_link_libraries(
```

Comment:
I think is more clear and easier to identify link errors if you separate the link to sfserr and slogger in different lines, i.e keep the original line and add a new one to link againt `slogger` library.